### ISSUE TITLE
Update translations after accessibility remediation updates

### ISSUE THREAD

**Issue by HaSistrunk** [RockefellerArchiveCenter/dimes#854]
## Is your feature request related to a problem? Please describe.
There will be a number of updates in DIMES in the coming weeks related to fixing accessibility issues identified in an internal audit. After this remediation work, we'll need to add some new translations for any new text we've added.

## Describe the solution you'd like
AFTER accessibility remediation work is complete, review HTML changes and add translations for any missing text. There will be new `aria-label` text and possibly other changes.

**Comment by HaSistrunk** [comment]
To translate:
1. Limitations and fees information
2. Indicates a required form field
3. Company Email
4. Menu
